crates/sui-source-validation-service/src/lib.rs

```diff
@@ -174,16 +174,30 @@ pub async fn initialize(
     dir: &Path,
 ) -> anyhow::Result<()> {
     clone_repositories(config, dir).await?;
-    verify_packages(context, vec![]).await?;
+    verify_packages(context, config, dir).await?;
     Ok(())
 }
 
 pub async fn verify_packages(
     context: &WalletContext,
-    package_paths: Vec<PathBuf>,
+    config: &Config,
+    dir: &Path,
 ) -> anyhow::Result<()> {
-    for p in package_paths {
-        verify_package(context, p).await?
+    let mut tasks = vec![];
+    for p in &config.packages {
+        let repo_name = repo_name_from_url(&p.repository)?;
```

Comment:
Doesn't have to be for this PR, but note that there's already an implementation in the move repo for converting a Git URL + sub-directory into a local path name, it might be worth standardising on that?

https://github.com/MystenLabs/sui/blob/c4493953deb182df1daa288f1d41efbb5bae76d2/external-crates/move/tools/move-package/src/resolution/mod.rs#L103-L114

Reply:
sounds good, going to look into this in a follow up!